cvat/apps/engine/views.py

```diff
@@ -132,32 +133,34 @@ def about(request):
         summary='Returns all files and folders that are on the server along specified path',
         parameters=[
             OpenApiParameter('directory', description='Directory to browse',
+                location=OpenApiParameter.QUERY, type=OpenApiTypes.STR),
+            OpenApiParameter('search', description='Search for specific files',
                 location=OpenApiParameter.QUERY, type=OpenApiTypes.STR)
         ],
         responses={
             '200' : FileInfoSerializer(many=True)
         })
     @action(detail=False, methods=['GET'], serializer_class=FileInfoSerializer)
     def share(request):
-        param = request.query_params.get('directory', '/')
-        if param.startswith("/"):
-            param = param[1:]
-        directory = os.path.abspath(os.path.join(settings.SHARE_ROOT, param))
+        directory_param = request.query_params.get('directory', '/')
+        search_param = request.query_params.get('search', '')
+
+        if directory_param.startswith("/"):
+            directory_param = directory_param[1:]
+
+        directory = (Path(settings.SHARE_ROOT) / directory_param).absolute()
 
-        if directory.startswith(settings.SHARE_ROOT) and os.path.isdir(directory):
+        if str(directory).startswith(settings.SHARE_ROOT) and directory.is_dir():
             data = []
-            content = os.scandir(directory)
-            for entry in content:
-                entry_type = None
-                entry_mime_type = None
+            for entry in (directory.iterdir() if not search_param else (f for f in directory.iterdir() if f.name.startswith(search_param))):
```

Comment:
I would suggest moving this iterator to a separate variable it's tough to read such code, two "for" and two "if-else" statements in one line of code  🤯